src/java-interop/java-interop-gc-bridge-mono.c

```diff
@@ -155,11 +160,36 @@ ji_realpath (const char *path)
 	if (path == NULL)
 		return NULL;
 
+#ifndef WINDOWS
 	char   *rp = realpath (path, NULL);
 	if (rp == NULL) {
 		return strdup (path);
 	}
 	return rp;
+#else
+#define BUFSIZE 4096
+
+	wchar_t *wpath = utf8_to_utf16 (path);
+	char *fullpath;
+	wchar_t buffer [BUFSIZE];
+	DWORD retval = GetFullPathNameW (wpath, BUFSIZE, buffer, NULL);
```

Comment:
In a world where we constantly complain about how short MAX_PATH is, we should do this properly. :-)

```c
WCHAR *wpath = NULL;
char *fullpath = NULL;
WCHAR* wfullpath = NULL;
do {
    path = utf8_to_utf16 (path);
    if (path == NULL)
        break;
    DWORD pathLen = GetFullPathNameW (wpath, 0, NULL, NULL);
    if (pathLen == 0)
        break;
    wfullpath = xmalloc (pathLen*sizeof(WCHAR));
    if (wfullpath == NULL)
        break;
    DWORD pathLen2 = GetFullPathNameW (wpath, pathLen, wfullpath, NULL);
    if (pathLen2 != pathLen) // Uh...wat?
        break;
    fullpath = utf16_to_utf8 (wfullpath);
} while (FALSE);
free (wpath);
free (wfullpath);
return fullpath;
```

Reply:
I don't see how that handle problem with `MAX_PATH` better compared to the code of this PR? It also fallbacks to allocate larger buffer when the original buffer is not long enough. Besides that, it is not documented whether that would even work as the documentation doesn't state what would happen when the 3rd argument is NULL. So it might fail, because it clearly states the 4th argument is allowed to be NULL, but doesn't say that for 3rd. Thus it looks better to me to use the stack buffer first.

To handle longer path than `MAX_PATH` we need path with `\\?\` prefix for the 1st argument. I assumed we already handle that and so the `path` passed to the ji_realpath already contains it?

Reply:
OTOH, in this example https://msdn.microsoft.com/en-us/library/windows/desktop/aa364989(v=vs.85).aspx the `GetShortPathName` is used that way, so I think it is OK to pass NULL to 3rd argument.

I even had it like that in the first iteration of the patch, but then noticed the missing NULL value case in the documentation and decided to use the stack allocated buffer to be safe :-)